### PR TITLE
Added mechanism to delete artifacts from GCS repository using MLFlow garbage collection

### PR DESCRIPTION
**Pull request by ghost**
## What changes are proposed in this pull request?

To delete artifacts from Google Cloud Storage, and tests to check if the function to delete the artifacts executes.

## How is this patch tested?

1. By having the backend store as a SQLite database - I have tested it by having a backend store as SQLite, by creating an experiment and few runs, while storing the artifacts in GCS. Then, I would delete the runs, and execute MLFlow garbage collection ([gc](https://mlflow.org/docs/latest/cli.html#mlflow-gc)).
2. In a production environment - I have tested it on a production environment, as well. An MLFlow instance was installed and running on a Kubernetes cluster. The backend store was MySQL database. I created an experiment and few runs, while storing the artifacts in GCS. Then, I would delete the runs, and execute MLFlow garbage collection ([gc](https://mlflow.org/docs/latest/cli.html#mlflow-gc)) in the cluster environment.
## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

The user can delete artifacts from GCS with MLFlow garbage collection.

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [x] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/store/artifact/gcs_artifact_repo.py [hunk 1]
```diff
@@ -140,4 +140,14 @@ def _download_file(self, remote_file_path, local_path):
...
+        if artifact_path:
+            dest_path = posixpath.join(dest_path, artifact_path)
+
+        gcs_bucket = self._get_bucket(bucket_name)
+        try:
+            blobs = gcs_bucket.list_blobs(prefix=f"{dest_path}")
+            for blob in blobs:
+                blob.delete()
+        except Exception:
+            raise MlflowException("Run not found")
```
Comment (dbczumar) [inline mlflow/store/artifact/gcs_artifact_repo.py:153]: @aditya-iyengar-rtl-de Can we remove the generic exception hander and instead let the underlying exception throw / propagate? This should make it easier for users to debug issues.

**Review by dbczumar — APPROVED** [review]
@aditya-iyengar-rtl-de Thanks so much for filing this PR! I've approved it with one small suggestion - https://github.com/mlflow/mlflow/pull/5811/files#r864033151. Happy to merge once that's addressed!

File: mlflow/store/artifact/gcs_artifact_repo.py (hunk 1 above)
Comment (aditya-iyengar-rtl-de) [reply mlflow/store/artifact/gcs_artifact_repo.py]: @dbczumar : I have removed the exception handler.

**Comment by dbczumar** [comment]
Thanks @aditya-iyengar-rtl-de ! Merging!

**Comment by aditya-iyengar-rtl-de** [comment]
Thanks @dbczumar :)

**Pull request merged**